README.md

```diff
@@ -55,8 +69,222 @@ documentation for more details.
+arn of your newly created IAM role will be exposed as a `CfnOutput`.
+
+To utilize this construct, create a separate CDK stack with the following code
+and `cdk deploy`:
+
+```ts
+import { AwsOidc } from 'cdk-pipelines-github';
+import { App, Construct, Stack, StackProps } from 'aws-cdk-lib';
+
+class MyAwsOidcRole extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const provider = new AwsOidc(this, 'oidc-role', {
+      repoString: 'myUser/myRepo',
+    };
+  }
+}
+
+const app = new App();
+new MyAwsOidcRole(app, 'MyAwsOidcRole');
+app.synth();
+```
+
+Note: If you have previously created the GitHub identity provider with url
+`https://token.actions.githubusercontent.com`, the above example will fail
+because you can only have one such provider defined per account. In this
+case, you must provide the already created provider into your `AwsOidc`
+construct via the `provider` property.
+
+```ts
+class MyAwsOidcRole extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const provider = new AwsOidc(this, 'oidc-role', {
+      repoString: 'myUser/myRepo',
+      provider: iam.OpenIdConnectProvider.fromOpenIdConnectProviderArn(
+        this,
+        'github',
+        'arn:aws:iam::<account-id>:oidc-provider/token.actions.githubusercontent.com',
+      ),
+    });
+  }
+}
+```
+
+#### Manual Setup
+
+In a nutshell, the IAM role you provide must reference the GitHub OIDC identity
+provider as a trusted entity. You must also set up a trust relationship between
+the IAM role and your GitHub repository. For a step-by-step tutorial on how to
+set this up, see
+[Configuring OpenID Connect in AWS](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services).
+
+In addition to setting up a relationship between GitHub and AWS, the IAM role 
+must also have permissions to assume CDK bootstrapped IAM roles and permissions 
+to access ECR repositories (if you plan on referencing Docker assets in your 
+workflow).
+
+Here is a minimum set of permissions for the IAM role: 
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Condition": {
+        "ForAnyValue:StringEquals": {
+          "iam:ResourceTag/aws-cdk:bootstrap-role": [
+            "deploy",
+            "lookup",
+            "file-publishing",
+            "image-publishing"
+          ]
+        }
+      },
+      "Action": "sts:AssumeRole",
+      "Resource": "*",
+      "Effect": "Allow"
+    },
+    {
+      "Action": "ecr:GetAuthorizationToken",
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+### Using Docker in the Pipeline
+
+You can use Docker in GitHub Workflows in a similar fashion to CDK Pipelines,
+except you cannot run Docker in the self-mutate project because there is no
```

Comment:
If it doesn't apply, better to leave it out than confuse the user. They don't need to know about self-mutation when using this. Don't tell them there's a thing they don't need to know about! 😀